File: examples/typescript/03_search.ts
import { ZeroEntropy } from 'zeroentropy';

const zclient = new ZeroEntropy();
const COLLECTION_NAME = "example_contracts";

async function main() {
    // Query for top documents
    console.log("=== Top Documents ===");
    const docResponse = await zclient.queries.topDocuments({
        collection_name: COLLECTION_NAME,
        query: "What are the payment terms?",
        k: 5,
        include_metadata: true,
    });

    for (const doc of docResponse.results) {
        console.log(`${doc.path} (score: ${doc.score.toFixed(4)})`);
        if (doc.metadata) {
            console.log(`  Tags: ${doc.metadata["list:tags"] || []}`);
        }
    }

    // Query for snippets with metadata filter
    console.log("\n=== Top Snippets (filtered) ===");
    const snippetResponse = await zclient.queries.topSnippets({
        collection_name: COLLECTION_NAME,
        query: "confidential information requirements",
        k: 3,
        precise_responses: true,
        filter: {
            "list:tags": { "$in": ["nda", "confidential"] },
        },
    });

    for (const snippet of snippetResponse.results) {
        console.log(`\n${snippet.path} [pages ${snippet.page_span}] (score: ${snippet.score.toFixed(4)})`);
        console.log(snippet.content?.slice(0, 200) + "...");
    }
}

main().catch(console.error);

File: examples/python/03_search.py
from zeroentropy import ZeroEntropy

zclient = ZeroEntropy()
COLLECTION_NAME = "example_contracts"

# Query for top documents
print("=== Top Documents ===")
response = zclient.queries.top_documents(
    collection_name=COLLECTION_NAME,
    query="What are the payment terms?",
    k=5,
    include_metadata=True,
)

for doc in response.results:
    print(f"{doc.path} (score: {doc.score:.4f})")
    if doc.metadata:
        print(f"  Tags: {doc.metadata.get('list:tags', [])}")

# Query for snippets with metadata filter
print("\n=== Top Snippets (filtered) ===")
response = zclient.queries.top_snippets(
    collection_name=COLLECTION_NAME,
    query="confidential information requirements",
    k=3,
    precise_responses=True,
    filter={
        "list:tags": {"$in": ["nda", "confidential"]},
    },
)

for snippet in response.results:
    print(f"\n{snippet.path} [pages {snippet.page_span}] (score: {snippet.score:.4f})")
    print(snippet.content[:200] + "...")
